From bbf5b2759cff0c65b47fd6dbe5fe7341f205ad03 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Subject: [PATCH] src/pppoe.h: fix build with musl libc

musl libc defines its own struct ethhdr that conflicts with the kernel
one. The kernel headers provide a way to suppress its struct ethhdr. For
that to work the libc headers must precede the kernel. Move the kernel
linux/if_ether.h include below libc netinet/if_ether.h.
(Backported from buildroot package/rp-pppoe.)

--- a/src/pppoe.h
+++ b/src/pppoe.h
@@ -117,10 +117,6 @@ typedef unsigned long UINT32_t;
 #error Could not find a 32-bit integer type
 #endif
 
-#ifdef HAVE_LINUX_IF_ETHER_H
-#include <linux/if_ether.h>
-#endif
-
 #include <netinet/in.h>
 
 #ifdef HAVE_NETINET_IF_ETHER_H
@@ -134,6 +130,10 @@ typedef unsigned long UINT32_t;
 #endif
 #endif
 
+#ifdef HAVE_LINUX_IF_ETHER_H
+#include <linux/if_ether.h>
+#endif
+
 
 
 /* Ethernet frame types according to RFC 2516 */
-- 
2.33.0

